data-access/mandataris.ts

```diff
@@ -403,3 +486,122 @@ export async function updatePublicationStatusOfMandataris(
     );
   }
 }
+
+async function findPerson(mandatarisId: string): Promise<string | undefined> {
+  const searchQuery = `
+    PREFIX person: <http://www.w3.org/ns/person#>
+    PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+    PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
+    PREFIX org: <http://www.w3.org/ns/org#>
+    PREFIX dct: <http://purl.org/dc/terms/>
+    PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+    PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+
+    SELECT DISTINCT ?person
+    WHERE {
+      GRAPH ?graph {
+        ?mandataris a mandaat:Mandataris;
+            mu:uuid ${sparqlEscapeString(mandatarisId)};
+            mandaat:isBestuurlijkeAliasVan ?person.
+      }
+
+      FILTER NOT EXISTS {
+        ?graph a <http://mu.semte.ch/vocabularies/ext/FormHistory>
+      } 
+    }
+  `;
+
+  const results = await querySudo(searchQuery);
+  const first = findFirstSparqlResult(results);
+
+  return first?.person.value;
+}
+
+async function findCurrentFractieInPeriod(
+  mandatarisId: string,
+  period: string | undefined,
+): Promise<string | null> {
+  const escapedBeeindigdState = sparqlEscapeUri(MANDATARIS_STATUS.BEEINDIGD);
+  const bestuursperiode = sparqlEscapeUri(period);
+  const searchQuery = `
+    PREFIX person: <http://www.w3.org/ns/person#>
+    PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+    PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
+    PREFIX org: <http://www.w3.org/ns/org#>
+    PREFIX dct: <http://purl.org/dc/terms/>
+    PREFIX mu: <http://mu.semte.ch/vocabularies/core/>
+    PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+
+    SELECT DISTINCT ?fractie ?mandataris ?bestuurorgaanInTijd
+    WHERE {
+      GRAPH ?graph {
+        ?mandataris a mandaat:Mandataris;
+            mu:uuid ${sparqlEscapeString(mandatarisId)};
+            mandaat:isBestuurlijkeAliasVan ?person;
+            org:hasMembership ?member;
+            org:holds ?mandaat;
+            dct:modified ?lastModified;
+            mandaat:status ?mandatarisStatus.
+        ?member org:organisation ?fractie.
+        ?mandaat ^org:hasPost ?bestuurorgaanInTijd.
+        ?bestuursorgaanInTijd ext:heeftBestuursperiode ${bestuursperiode}.
+
+        OPTIONAL {
+          ?mandataris mandaat:einde ?endDate.
+        }
+      }
+
+      FILTER ( 
+        ?mandatarisStatus != ${escapedBeeindigdState} &&
+        ?lastModified <= ?safeEnd
+      )
+      FILTER NOT EXISTS {
+        ?graph a <http://mu.semte.ch/vocabularies/ext/FormHistory>
+      } 
+      BIND(IF(BOUND(?endDate), ?endDate,  ?lastModified) as ?safeEnd)
+    }
+  `;
+
+  const results = await querySudo(searchQuery);
+  const fracties = getSparqlResults(results);
+  console.log(fracties);
+  if (fracties.length === 0 || !fracties[0].fractie || !period) {
+    return null;
+  }
+
+  const doubleCheckInPeriod = await isInBestuursperiode(
```

Comment:
The bindings are just retunring the unique comibinations and that is not always right, added this doiuble check as than we are really sure the current mandataris does belong to the active bestuursperiode